lms/envs/common.py

```diff
@@ -1059,6 +1059,15 @@
     # .. toggle_creation_date: 2024-03-22
     # .. toggle_tickets: https://github.com/openedx/edx-platform/pull/33911
     'ENABLE_GRADING_METHOD_IN_PROBLEMS': False,
+
+    # .. toggle_name: FEATURES['DISABLE_DATES_TAB']
+    # .. toggle_implementation: DjangoSetting
+    # .. toggle_default: False
+    # .. toggle_description: Disables dates tab for all courses.
+    # .. toggle_use_cases: open_edx
+    # .. toggle_creation_date: 2024-04-15
+    # .. toggle_tickets: https://github.com/openedx/edx-platform/pull/34511
+    'DISABLE_DATES_TAB': False,
```

Comment:
Instead of an instance-wide setting, I think it makes more sense to make this configurable per course, by setting is_hideable = True in the DatesTab class, and creating a course app to configure it in case the pages and resources MFE is enabled (e.g. https://github.com/openedx/edx-platform/blob/a10fce33794e9c527d3f651408b16f1ad5b7fec9/lms/djangoapps/courseware/plugins.py#L48).